### PR TITLE
 Connector support for API Token Auth POC

### PR DESCRIPTION
**Pull request by mchernyavskaya**
#### Pre-Review Checklist
- [ ] Covered the changes with automated tests
- [ ] Tested the changes locally
- [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

## For Elastic Internal Use Only
- [ ] Considered corresponding documentation changes to [contribute separately](https://github.com/elastic/enterprise-search-pubs#contribute-docs-changes-for-product-changes)
- [ ] New configuration settings added in this PR follow the [official guidelines](https://github.com/elastic/ent-search/blob/main/doc/enterprise-search-config.md)
- [ ] [Manual test steps](https://github.com/elastic/connectors/blob/main/docs/INTERNAL.md#minimal-manual-tests)  are successful
- [ ] Enterprise Search builds successfully with a gem built from this PR's branch.
  - _Link the ent-search PR here_

### CODE REVIEW

File: config/connectors.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# general metadata
```
Comment (tarekziade) [inline config/connectors.yml:1]: do we want to commit this file? (maybe we should add it in `.gitignore`

File: lib/connectors_sdk/base/authorization.rb [hunk 2]
```diff
@@ -70,6 +70,10 @@ def oauth_scope
           raise 'Not implemented for this connector'
         end
 
+        def basic_auth_token(params)
+          raise 'Not implemented for this connector'
```
Comment (tarekziade) [inline lib/connectors_sdk/base/authorization.rb:74]: should we raise `NotImplementedError` ?

File: lib/connectors_sdk/confluence_cloud/extractor.rb [hunk 3]
```diff
@@ -52,7 +52,8 @@ def yield_permissions(source_user_id)
       def download(item)
         content = item[:content]
         parent_id = content.dig('container', 'id')
-        client.download("#{client.base_url}/wiki/rest/api/content/#{parent_id}/child/attachment/#{content['id']}/download").body
+        wiki_path = client.base_url.downcase.include?('/wiki') ? '' : '/wiki'
```
Comment (tarekziade) [inline lib/connectors_sdk/confluence_cloud/extractor.rb:55]: what's the goal of this change? it seems unrelated to the Auth POC

File: spec/connectors_sdk/confluence/custom_client_spec.rb [hunk 4]
```diff
@@ -90,6 +101,37 @@
     expect(content_search_request).to have_been_requested
   end
 
+  it 'applies correct middleware for token auth' do
```
Comment (tarekziade) [inline spec/connectors_sdk/confluence/custom_client_spec.rb:104]: awesome!

**Review by tarekziade — COMMENTED** [review]
Design question

We're adding everywhere `basic_auth_token` and now each class has to deal with `access_token` and `basic_auth_token`

I am wondering if we should not move the complexity away from options passing.
e.g. define a new `AuthenticationToken` class along with a type (basic, bearer)  that can be used to forge our requests and provide all helpers around the actual raw token. I am worried that we are carrying around auth details that could be encapsulated.

File: lib/connectors_sdk/confluence_cloud/extractor.rb (hunk 3 above)
Comment (mchernyavskaya) [reply lib/connectors_sdk/confluence_cloud/extractor.rb:55]: It is indeed related. For the Confluence Cloud, the base API URL is different for OAuth and API Token Auth. And it already includes `wiki` in the second case. So I had to add this to not add it again.

**Comment by mchernyavskaya** [comment]
Closing as not relevant. The BYOC work makes this obsolete.

**Comment by mchernyavskaya** [comment]
partially the change was merged in [this PR](https://github.com/elastic/connectors/pull/118) by Sean

**Pull request closed without merging**